src/Parser/Mage/Fire/Normalizers/PyroclasmBuff.js

```diff
@@ -0,0 +1,38 @@
+import SPELLS from 'common/SPELLS';
+
+import EventsNormalizer from 'Parser/Core/EventsNormalizer';
+
+class PyroclasmBuff extends EventsNormalizer {
+  /**
+   * @param {Array} events
+   * @returns {Array}
+   */
+  //Ensures that the  ApplyBuff, RefreshBuff, and RemoveBuff events are not occuring before the pyroblast events... so the buff doesnt get applied, removed, or refreshed before the pyroblast actually casts
+  normalize(events) {
```

Comment:
Please consider adding a test for any normalizers you make. They are really important to have working well and any issues now or introduced later could make analysis extremely challenging.

Reply:
Is there an example of this somewhere? Im not entirely sure what that means or what it does. 

Sorry, Javascript is still pretty new to me .

Reply:
What module did you base this on? There are a bunch with tests, like the LightOfDawn.test.js file.

Reply:
i dont remember, i made one of them a long time ago and just kept duplicating that one. Ill go looking for that one though, thanks.

Reply:
Do note it's not required for this PR to be merged, it's for your own convenience down the line

Reply:
I took a look at it, ill have to spend some time working with it to get an idea of how it works and such. Ill get them added for all of the normalizers im using (4 of them in total) in a later PR